node/src/components/small_network/tasks.rs

```diff
@@ -63,6 +63,13 @@ use crate::{
 /// successfully handed over to the kernel for sending.
 pub(super) type MessageQueueItem<P> = (Arc<Message<P>>, Option<AutoClosingResponder<()>>);
 
+struct HandshakeOutcome {
```

Comment:
Could you add docs on the struct and its members? :sparkles: I try to be diligent about this, even (or especially :)) internal structs.